feat: generate scala classes and traits by exploring an ast.

JonathanGeoffroy opened the pull request:
Generate scala files in a directory, with a package provided by user.
Generate each scala trait contained by the ast.
For each scala class, generate not only a class and its fields, but also a companion object, which contains an apply method for each field.
Provide an Example class, with few basic html markers.

JonathanGeoffroy committed:
feat: generate scala classes and traits by exploring an ast.

Generate scala files in a directory, with a package provided by user.
Generate each scala trait contained by the ast.
For each scala class, generate not only a class and its fields, but also a companion object, which contains an apply method for each field.
Provide an Example class, with few basic html markers.

rouvoy commented on src/main/scala/fr/inria/hopla/generator/ASTClassGenerator.scala:
```diff
@@ -0,0 +1,236 @@
+  /**
+   * Generate a class and its fields and attributes
+   */
+  class ClassGenerator {
```
This class can be replaced by an object

rouvoy commented on src/main/scala/fr/inria/hopla/generator/ASTClassGenerator.scala:
```diff
@@ -0,0 +1,236 @@
+  /**
+   * Generate a companion object and its apply methods
+   */
+  class CompanionGenerator extends ScalaGenerator(astClass) {
```
This class can be replaced by an object.

rouvoy commented on src/main/scala/fr/inria/hopla/generator/ASTFileGenerator.scala:
```diff
@@ -0,0 +1,19 @@
+   * @param outputDirectory the path to the directory where to create scala files.
+   */
+  def generate(outputDirectory: String, packageName: String): Unit
+}
```
Missing empty line.

JonathanGeoffroy replied:
The last commit resolves this issue, but it doesn't appear in the github preview.
However, it may be cleaner to squash the three commits, but I'm not sure it's possible without closing this pull request and opening a new one...

rouvoy replied:
You can close this one and reopen a new request.

rouvoy commented on src/main/scala/fr/inria/hopla/generator/PrintGenerator.scala:
```diff
@@ -0,0 +1,104 @@
+    }
+    writer.close()
+  }
+}
```
Missing empty line.

JonathanGeoffroy committed:
feat: generate scala classes and traits by exploring an ast.

Generate scala files in a directory, with a package provided by user.
Generate each scala trait contained by the ast.
For each scala class, generate not only a class and its fields, but also a companion object, which contains an apply method for each field.
Provide an Example class, with few basic html markers.

JonathanGeoffroy committed:
feat: generate scala classes and traits by exploring an ast.

Generate scala files in a directory, with a package provided by user.
Generate each scala trait contained by the ast.
For each scala class, generate not only a class and its fields, but also a companion object, which contains an apply method for each field.
Provide an Example class, with few basic html markers.
